tests/ui/traits/const-traits/tilde-const-in-struct-args.rs

```diff
@@ -0,0 +1,21 @@
+//@ compile-flags: -Znext-solver
```

Comment:
This is the known-bug for `tests/crashes/119924-6.rs`. We now silently accept `~const Trait` in illegal positions when they're nested in legal positions, because AST validation is not smart enough. I don't think we need to fix this immediately, though I think it would be nice if we made validation more robust.